### PR TITLE
Import this week in Rails

### PR DESCRIPTION
**Pull request by vipulnsward**

### CODE REVIEW

File: _posts/2023-09-27-this-week-in-rails.markdown [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+```ruby
+class Post < ActiveRecord::Base 
+  attribute :topic, :string 
+  enum topic: %i[science tech engineering math]
+end
+```
+
+[Handle negative numbers in NumberToHumanSizeConverter](https://github.com/rails/rails/pull/49791)
+This Pull Request adds support for `number_to_human_size` to handle negative numbers-
```
Comment (gregmolnar) [inline _posts/2023-09-27-this-week-in-rails.markdown:33]: The sentence ends with a `-`

File: _posts/2023-09-27-this-week-in-rails.markdown [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+```ruby
+params = ActionController::Parameters.new(tags: "ruby,rails,,web")
+params.extract_value(:tags) # => ["ruby", "rails", "", "web"]
+```
+
+This helps to fix a bug where using `url_helpers` generated invalid urls if a records second composite primary key is empty.
+
+[Ignore invalid PRIMARY KEY values in ActiveRecord::Persistence.delete method](https://github.com/rails/rails/pull/49738)
+
+When some invalid primary key values are passed to `ActiveRecord::Persistence.delete` method, ActiveRecord still performed a `DELETE` query even if it is not needed. For example,
```
Comment (gregmolnar) [inline _posts/2023-09-27-this-week-in-rails.markdown:69]: "ActiveRecord still" should be "Active Record still"

**Comment by gregmolnar** [comment]
@vipulnsward other than the 2 comments I made, it is looking good.

**Comment by vipulnsward** [comment]
@gregmolnar thanks fixed those and some more things

**Comment by gregmolnar** [comment]
@vipulnsward looking good to me, just please squash the commits before merging.

**Pull request merged**